src/components/canvas/Canvas.vue

```diff
@@ -62,9 +67,31 @@ export default class Canvas extends Vue {
     this.engine.events.calculated.addListener(this, () => {
       // Building IR
       const currEditorSave = saveEditor(this.editorModel);
-      istateToGraph(currEditorSave.state);
+      const irGraph = istateToGraph(currEditorSave.state);
+      const errors = Canvas.errorMap(check(irGraph));
+      this.setErrorsMap(errors);
+      if (errors.size !== 0) {
+        console.log(`Found ${errors.size} errors`);
+      }
     });
   }
+
+  private static errorMap(errors: IrError[]): Map<string, IrError[]> {
```

Comment:
It would make sense to me to move this method into some Typescript file? It feels like it doesn't really belong to the Component, and this would be the only static method in a Component in the project. Not sure about it though, @gidonkatten?

Reply:
at first it was inlined (took it out for visibility) I am not planning to use it anywhere else though. Maybe we can move it down the line

Reply:
Sure, let's do that!